File: loan-calculator-ui/src/app/models/loan-models.ts
export interface LoanCalculationRequest {
  LoanAmount: number;
  InterestRate: number;
  TermInYears: number;
}

export interface MonthlyPaymentDetail {
  PaymentNumber: number;
  Payment: number;
  Principal: number;
  Interest: number;
  RemainingBalance: number;
}

export interface LoanCalculationResult {
  Id?: string;  // Optional since it might not be included in the /payments response
  LoanAmount: number;
  InterestRate: number;
  TermInYears: number;
  MonthlyPayment: number;
  PaymentSchedule: MonthlyPaymentDetail[];
  CalculationDate?: Date;  // Optional since it might not be included in the /payments response
}

File: src/demo/Models/LoanModels.cs
using System;
using System.Collections.Generic;

namespace demo.Models
{
    public class LoanCalculationResult
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public decimal LoanAmount { get; set; }
        public decimal InterestRate { get; set; }
        public int TermInYears { get; set; }
        public decimal MonthlyPayment { get; set; }
        public List<MonthlyPaymentDetail> PaymentSchedule { get; set; } = new List<MonthlyPaymentDetail>();
        public DateTime CalculationDate { get; set; } = DateTime.UtcNow;
        public string CalculationMethod { get; set; } = "Shpitzer"; // Default to Shpitzer
    }

    public class MonthlyPaymentDetail
    {
        public int PaymentNumber { get; set; }
        public decimal Payment { get; set; }
        public decimal Principal { get; set; }
        public decimal Interest { get; set; }
        public decimal RemainingBalance { get; set; }
    }

    public class LoanCalculationRequest
    {
        public double LoanAmount { get; set; }
        public double InterestRate { get; set; }
        public int TermInYears { get; set; }
        public string CalculationMethod { get; set; } = "Shpitzer"; // Default to Shpitzer
    }
}